libbpf-tools/tcpretrans.c

```diff
@@ -0,0 +1,243 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2020 Anton Protopopov
+// Copyright (c) 2021 Red Hat, Inc.
+//
+// Based on tcpconnect.c by Anton Protopopov and
+// tcpretrans(8) from BCC by Brendan Gregg
+#include <sys/resource.h>
+#include <arpa/inet.h>
+#include <argp.h>
+#include <signal.h>
+#include <limits.h>
+#include <unistd.h>
+#include <time.h>
+#include <bpf/bpf.h>
+#include "tcpretrans.h"
+#include "tcpretrans.skel.h"
+#include "trace_helpers.h"
+#include "map_helpers.h"
+
+#define warn(...) fprintf(stderr, __VA_ARGS__)
+
+const char *argp_program_version = "tcpretrans 0.1";
+const char *argp_program_bug_address =
+	"https://github.com/iovisor/bcc/tree/master/libbpf-tools";
+static const char argp_program_doc[] =
+	"\ntcpretrans: Trace TCP retransmits\n"
+	"\n"
+	"EXAMPLES:\n"
+	"    tcpretrans             # display all TCP retransmissions\n"
+	// TODO "    tcpconnect -c          # count occurred retransmits per flow\n"
+	// TODO "    tcpconnect -l      	# include tail loss probe attempts\n"
+	;
+
+const char* TCPSTATE[] = {
+	"ESTABLISHED",
+	"SYN_SENT",
+	"SYN_RECV",
+	"FIN_WAIT1",
+	"FIN_WAIT2",
+	"TIME_WAIT",
+	"CLOSE",
+	"CLOSE_WAIT",
+	"LAST_ACK",
+	"LISTEN",
+	"CLOSING",
+	"NEW_SYN_RECV"};
+
+
+static volatile sig_atomic_t hang_on = 1;
```

Comment:
Existing tools use `exiting` for this purpose. I think it's good to follow that.

Reply:
syscount.c and tcpconnect. also use the 'hang_on' pattern.